src/OAuth2.php

```diff
@@ -558,7 +558,7 @@ public function buildFullAuthorizationUri(array $config = null)
     }
 
     // Construct the uri object; return it if it is valid.
-    $result = $this->authorizationUri;
+    $result = clone $this->authorizationUri;
```

Comment:
without this, if you build the authorization URI twice in one session, the querystring parameters will be added twice